### ISSUE TITLE
Cannot push to mirror

### ISSUE THREAD

**Issue by JoshuaCurry** [gogits/gogs#3892]
My friend as a git server. (gogs)
I also have a server running gogs.
When I setup a mirror of one of his repos (on which I am a collaborator) on to my server, I can pull the mirrored repo perfectly fine, but cannot push and have it mirrored back to his server.

The error I get is a 403 with error 'remote: mirror repository is read-only.'

Surely it should be possible to setup bidirectional mirrors; or is the gogs functionality limited to pull-only?

**Comment by afdev82** [comment]
As far as I know, in general the mirror functionality (of git) is intended in one way only.
I just finished to setup a bidirectional mirror with gogs and other remote repositories and I did in this way:
- create a normal bare repository
- add the other repository as remote, eg. `git remote add origin repo_url`
- access the repository directory with ssh
- modify the config file in the root of the repo, adding the `mirror = true` option under the section of the remote, modify also the fetch option to include all the refs, in the end it should like  
```
[remote "origin"]
 url    = remote_repo_url
 fetch  = +refs/*:refs/*
 mirror = true
```
- add a post-receive hook in the repository's settings page, like this  
```
git remote prune origin
git push --quiet origin
```
In this way, everytime you push something into your repository, the script will update (push) in the remote repository.
If you do exactly the same for the remote repository, just setting your repository as remote, it will work in the other direction, updating your repository everytime the other repository is updated.
